### DISCUSSION TITLE
zlib `1.5.6` removed from releases

### DISCUSSION THREAD

**Discussion by m-vz** [Homebrew/discussions#5246] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.15-53-g02a0ea8
ORIGIN: https://github.com/Homebrew/brew
HEAD: 02a0ea84492f07d5d3522ba45958166ec0d5481b
Last commit: 13 hours ago
Core tap JSON: 30 Mar 13:15 UTC
Core cask tap JSON: 30 Mar 13:15 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: nvim
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 13.6.3-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: No Cask quarantine support available: unknown reason.
```

### Description of issue

I'm not familiar with how the distributed version of a brew package is determined, so feel free to close this if it turns out to be irrelevant.

The `1.5.6` release of zlib is not listed on the [zlib releases page](https://github.com/facebook/zstd/releases) anymore. There is also a discussion about this here: https://github.com/facebook/zstd/issues/4004

Is the published version of the zlib homebrew package determined from the [tag](https://github.com/facebook/zstd/releases/tag/v1.5.6) that still exists or from the release?

**Comment by SMillerDev** [comment, marked as answer]
We'll keep an eye on it, but generally unless it's a known security issue. We wait for upstream to explain why it's gone.

**Reply by m-vz** [reply]
Thanks, that makes sense.

**Comment by ZhongRuoyu** [comment]
Upstream have explained why the release was remade in https://github.com/facebook/zstd/issues/4004#issuecomment-2028441787. The tag still points to the same commit, and the tarball checksum stays the same. So, nothing will change in `homebrew/core`.

(The package you're referring to is Facebook's `zstd`; `zlib` is a different package.)
